### ISSUE TITLE
A new, clever way to use foci

### ISSUE THREAD

**Issue by yagomont** [Azanor/thaumcraft-beta#1343]
I've been pondering about the effects available to foci in Thaumcraft 6, and reached the conclusion that here to the final release new effects will be present in the mod. Something that I assume is frequently thought about are effects that apply buffs to its target. It's a nice idea that has been explored by other magic mods, but, there's another way to handle buffs.

Let's suppose that focus buffs are implemented. Typically, a player would have to frequently switch over to another focus to reapply their speed buff, and so on. Such a focus would typically have the root as target, directly linked to the speed-giving effect. Such a simple, root-effect focus would be simple enough to not require the complicated device that is a gauntlet.

My idea is that such simple foci could be attached to a sufficiently magical ring, say, a thaumium ring. Let's remember we only have the cloudstepper ring right now. The focus would automatically channel vis, at a reduced price, passively, to constantly keep the player with such buff.

This idea, to me, sounds like something our Thaumaturge could be able to think of, once they get far enough in Auromancy.

**Comment by TechnoTadhg** [comment]
Makes sense, having a bauble you can open like the focus pouch and place x amount of passive effect foci into it, although it might be a bit op if done poorly.
Also on buffs in general - Most mods do it very poorly, like sure some like botania do it alright but most mods just toss vanilla buffs on you and it just seems lazy. Look at all the "kids first mods" that just add armour that adds buffs.
Unless these are new buffs with unique qualities I don't see the point, I'd just rather stuff like the cloudstepper ring that add new abilities, and if we get a lot hopefully a method to increase the number of bauble slots to accommodate, ~~and fix the problem of mods fighting for bauble slots with each other~~.

**Comment by yagomont** [comment]
My opinion, but, I think vanilla buffs are underappreciated quality-of-life changers. Haste, resistance, regeneration... are unique on their own, considering their versatility. Using foci crafting as a base, with an infusion recipe to add in the loss and difficulty factor, and the immutability of the ring's focus (the focus would also need to shrink, hence why we need to use infusion magic, methinks is an argument) would kind of repel the kids first tendency, since the mods that do that usually make recipes for X armor using easy-to-get Y item for a full set buff.

**Comment by TechnoTadhg** [comment]
I'd say it would still come off as lazy, compared to the rest of the mod. Also, it would kinda invalidate beacons, we don't want those to be made useless since you have the buffs you want already.
I still think more items like the cloudstepper ring would be the way to go, it acts as jump boost but would also stack with a jump boost beacon. (at least for the first jump, haven't tested the double jump)
